app/src/main/java/de/danoeh/antennapod/ui/CoverLoader.java

```diff
@@ -1,132 +1,242 @@
-        private final boolean textAndImageCombined;
+    private void loadWithImageModel(RequestOptions options) {
+        RequestBuilder<Drawable> builder = GlideApp.with(coverView)
+                .as(Drawable.class)
+                .load(imageModel)
+                .apply(options);
 
-        public CoverTarget(TextView fallbackTitle, ImageView coverImage, boolean textAndImageCombined) {
-            super(coverImage);
-            this.fallbackTitle = new WeakReference<>(fallbackTitle);
-            this.cover = new WeakReference<>(coverImage);
-            this.textAndImageCombined = textAndImageCombined;
-        }
+        // Add error handling for failed network requests
+        builder = builder.error(GlideApp.with(coverView)
+                .as(Drawable.class)
+                .load(new ErrorImageModel(imageModel.getFallbackText()))
+                .apply(options));
 
-        @Override
-        public void onLoadFailed(Drawable errorDrawable) {
-            setTitleVisibility(fallbackTitle.get(), true);
-        }
+        builder.into(coverView);
+    }
 
-        @Override
-        public void onResourceReady(@NonNull Drawable resource,
-                                    @Nullable Transition<? super Drawable> transition) {
-            ImageView ivCover = cover.get();
-            ivCover.setImageDrawable(resource);
-            setTitleVisibility(fallbackTitle.get(), textAndImageCombined);
+    /**
+     * Loads the image as a Bitmap with automatic error handling.
+     * On success, calls callback with the loaded bitmap.
+     * On error, automatically loads ErrorImageModel and calls callback with error bitmap.
+     * 
+     * @param callback BitmapCallback that receives the final bitmap (either success or error)
+     */
+    public void get(BitmapCallback callback) {
+        if (activity == null) {
+            throw new IllegalStateException("Activity context required for get() method");
         }
-
-        @Override
-        protected void onResourceCleared(@Nullable Drawable placeholder) {
-            ImageView ivCover = cover.get();
-            ivCover.setImageDrawable(placeholder);
-            setTitleVisibility(fallbackTitle.get(),  textAndImageCombined);
+        
+        RequestBuilder<android.graphics.Bitmap> builder = GlideApp.with(activity).asBitmap();
+        
+        if (imageModel != null) {
+            builder = builder.load(imageModel);
+        } else if (resource != 0) {
+            builder = builder.load(resource);
         }
-
-        static void setTitleVisibility(TextView fallbackTitle, boolean textAndImageCombined) {
-            if (fallbackTitle != null) {
-                fallbackTitle.setVisibility(textAndImageCombined ? View.VISIBLE : View.GONE);
-            }
+        
+        if (requestOptions != null) {
+            builder = builder.apply(requestOptions);
         }
+        
+        builder.into(new CustomTarget<android.graphics.Bitmap>() {
```

Comment:
Instead of a custom target, I think it is possible to give Glide a fallback builder directly. This would make the code easier to understand